documentation/modules/oauth/con-oauth-reauthentication.adoc

```diff
@@ -1,23 +1,41 @@
+This Kafka mechanism is used to enforce the expiry of authenticated sessions between OAuth clients and Kafka brokers after a defined period of time. 
+When a session expires, the client immediately starts a new session by reusing the existing connection rather than dropping it.
 
-When the re-authentication mechanism is enabled in the configuration of an `oauth` type listener, the broker's authenticated session expires when the access token expires.
-The client must then re-authenticate to the existing session by sending a new, valid access token to the broker, without dropping the connection.
+Session re-authentication is disabled by default. 
+To enable and configure it, you set a time value for `maxSecondsWithoutReauthentication` in the `oauth` listener configuration in the `Kafka` custom resource. 
+The same property is used to configure session re-authentication for OAUTHBEARER and PLAIN authentication. 
+For an example configuration, see xref:proc-oauth-authentication-broker-config-{context}[].
+
+Session re-authentication can be used with _fast local JWT_ or _introspection endpoint_ token validation.
+
+.Client re-authentication
+
+When the broker's authenticated session expires, the client must re-authenticate to the existing session by sending a new, valid access token to the broker, without dropping the connection.
 
 If token validation is successful, a new client session is started using the existing connection.
 If the client fails to re-authenticate, the broker will close the connection if further attempts are made to send or receive messages.
 Java clients that use Kafka client library 2.2 or later automatically re-authenticate if the re-authentication mechanism is enabled on the broker.
 
-You enable session re-authentication in the `Kafka` resource. Set the `maxSecondsWithoutReauthentication` property for a TLS listener with `type: oauth` authentication.
-Session re-authentication is supported for both types of token validation (fast local JWT and introspection endpoint). For an example configuration, see xref:proc-oauth-authentication-broker-config-{context}[].
+.Differences in session expiry for OAUTHBEARER and PLAIN
+
+There are some differences in how session expiry works when session re-authentication is configured for OAUTHBEARER and PLAIN authentication.
 
-For more information about the re-authentication mechanism, which was added in Kafka version 2.2, see link:https://cwiki.apache.org/confluence/display/KAFKA/KIP-368%3A+Allow+SASL+Connections+to+Periodically+Re-Authenticate[KIP-368^].
+* For OAUTHBEARER, the broker's authenticated session will expire at the configured `maxSecondsWithoutReauthentication`. 
+The session will expire earlier if the access token expires before the `maxSecondsWithoutReauthentication`. 
+
+* For PLAIN, the broker's authenticated session will expire at the configured `maxSecondsWithoutReauthentication`. 
```

Comment:
What if PLAIN config uses a long-lived access token?

Reply:
If the access token expires before the `maxSecondsWithoutReauthentication` then the re-authentication is triggered then. However, in that case the client will send an expired token as part of re-authentication because it has no mechanism in PLAIN to refresh the token. As a result the authentication will fail and the client will not be able to continue without configuring a new access token as the password, that is typically a manual step. This is where OAUTHBEARER would have no problem, and using clientId + secret over PLAIN would also have no problem.

Reply:
I don't think that matters.

Reply:
OK, I attempted to add an explanation of the behaviour.

> For PLAIN using the long-lived access token method:
> 
> * The broker's authenticated session will expire at the configured `maxSecondsWithoutReauthentication`. 
> 
> * Re-authentication will fail if the access token expires before the configured time. Although session re-authentication is attempted, PLAIN has no mechanism for refreshing tokens.